ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExchangeTransitionConfigurationTest.java

```diff
@@ -153,17 +153,33 @@ public void shouldAlwaysReturnResultsInHex() throws JsonProcessingException {
         new EngineExchangeTransitionConfigurationResult(Difficulty.ZERO, Hash.ZERO, 0L);
 
     assertThat(mockResult.getTerminalBlockNumberAsString()).isEqualTo("0x0");
-    assertThat(mockResult.getTerminalTotalDifficultyAsString())
-        .isEqualTo(Difficulty.ZERO.toHexString());
+    assertThat(mockResult.getTerminalTotalDifficultyAsString()).isEqualTo("0x0");
     assertThat(mockResult.getTerminalBlockHashAsString()).isEqualTo(Hash.ZERO.toHexString());
 
     String json = mapper.writeValueAsString(mockResult);
     var res = mapper.readValue(json, Map.class);
     assertThat(res.get("terminalBlockNumber")).isEqualTo("0x0");
     assertThat(res.get("terminalBlockHash"))
         .isEqualTo("0x0000000000000000000000000000000000000000000000000000000000000000");
```

Comment:
The leading 0s need to be stripped. The spec says:
```
Values of a field of QUANTITY type MUST be encoded as a hexadecimal string with a 0x prefix and the leading 0s stripped (except for the case of encoding the value 0) matching the regular expression ^0x(?:0|(?:[a-fA-F1-9][a-fA-F0-9]*))$.
```

Reply:
terminalBlockHash is of type `DATA`, not `QUANTITY`. leading zeros are ok in those.

https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#TransitionConfigurationV1

Reply:
My bad. Then it's correct of course